### ISSUE TITLE
`CGRSolver` "Subspace" Preconditioner Broken

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#1918]
The following input file
```
molecule {
O
H 1 1.0 
H 1 1.0 2 104.5
}

set solver_precondition subspace
set solver_n_guess 3

gradient('mp2/cc-pvdz')
```
segfaults due to a bad access exception. While the segfault occurred on [this line](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L290), the real problem is earlier. The loop structure [assumes](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L280-L285) that `x_` and `b_` have the same number of elements. But the last method we ran [was `setup`](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L163-L164), which [clears `x_`](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L276).

Even then, `setup` appears to have its own, unrelated problem. The preconditioner submatrix `A_` is [populated using the results of a `products_x()` call](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L259-L272). But `x_` [consists of zero vectors](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/solver.cc#L251), so the submatrix guess procedure is just an overcomplicated way to write zero at present. That is probably very bad for the algorithm.

Making matters even worse, I have not been able to find a standard conjugate gradients preconditioner with a name similar to "SUBSPACE," so I cannot figure out how this algorithm was supposed to work.

In short, this code is broken, and I would say beyond repair. Can I eliminate all the code for the "SUBSPACE" preconditioner from the `CGRSolver`? We'd just fall back to the "JACOBI" preconditioner, which is the default anyways.

**Comment by robertodr** [comment]
I think the algorithm projects A in a subspace, inverts it exactly, and uses that as preconditioner. This said, I would not try to heal it and remove that code branch. Given that it never became the default, there's a high chance the cost of improving the preconditioner outweighed the gain in speed of convergence.
